app/components/domains-table.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Table,
+  Tr,
+  Th,
+  Thead,
+  Tbody,
+  TableContainer,
+  Td,
+  IconButton,
+  Flex,
+  Tooltip,
+  Card,
+  useToast,
+} from '@chakra-ui/react';
+import type { Record, RecordStatus } from '@prisma/client';
+import {
+  EditIcon,
+  DeleteIcon,
+  CheckCircleIcon,
+  RepeatIcon,
+  CopyIcon,
+  TimeIcon,
+  WarningIcon,
+} from '@chakra-ui/icons';
+
+interface DomainsTableProps {
+  domains: Record[];
+  onAction: (domaindID: number, action: DomainsTableAction) => void;
+}
+
+export type DomainsTableAction = 'EDIT' | 'DELETE' | 'RENEW';
+
+export default function DomainsTable(props: DomainsTableProps) {
+  const { domains, onAction } = props;
+
+  const toast = useToast();
+
+  function onCopyNameToClipboard(name: string) {
+    navigator.clipboard.writeText(name);
+    toast({
+      title: 'Name was copied to clipboard',
+      position: 'bottom-right',
+      status: 'success',
+    });
+  }
+
+  function renderDomainStatus(action: RecordStatus) {
+    switch (action) {
```

Comment:
Nit:

You can do this with an early return style:
```
if (action === 'active'){
  return (...)
}

if (action === 'error'){
  return (...)
}
```

Reason:
It is usually considered a bad practice to use a switch case, because there is too much risk of the execution overflowing to the next case ... just an unclean language element